cmd/controller/controller.go

```diff
@@ -94,13 +97,22 @@ func NewControllerCmd() *cobra.Command {
 			c.Logging = stringmap.Merge(c.CmdLogLevels, c.DefaultLogLevels)
 			cmd.SilenceUsage = true
 
+			if err := (&sysinfo.K0sSysinfoSpec{
+				ControllerRoleEnabled: true,
+				WorkerRoleEnabled:     c.SingleNode || c.EnableWorker,
+				DataDir:               c.K0sVars.DataDir,
+			}).RunPreFlightChecks(ignorePreFlightChecks); err != nil {
+				return err
```

Comment:
This will still error out, even if `ignorePreFlightChecks` is `true`, in case of any hard errors (i.e. `Error` has been called on the `Reporter`). Not sure if we want it like that.